### ISSUE TITLE
in regex-howto, improve example on grouping

### ISSUE THREAD

**Issue by fc9cca60-f582-41bb-a683-840c6b119ece** [python/cpython#74190]
BPO | [30004](https://bugs.python.org/issue30004)
--- | :---
Nosy | @akuchling, @ezio-melotti, @serhiy-storchaka, @Mariatta, @mandeep
PRs | <li>python/cpython#4443</li><li>python/cpython#4554</li><li>python/cpython#4555</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-11-25.05:03:50.427>
created_at = <Date 2017-04-06.04:40:42.518>
labels = ['expert-regex', 'type-feature', '3.7', 'docs']
title = 'in regex-howto, improve example on grouping'
updated_at = <Date 2017-11-25.07:03:29.823>
user = 'https://bugs.python.org/CristianBarbarosie'
```

bugs.python.org fields:
```python
activity = <Date 2017-11-25.07:03:29.823>
actor = 'serhiy.storchaka'
assignee = 'docs@python'
closed = True
closed_date = <Date 2017-11-25.05:03:50.427>
closer = 'Mariatta'
components = ['Documentation', 'Regular Expressions']
creation = <Date 2017-04-06.04:40:42.518>
creator = 'Cristian Barbarosie'
dependencies = []
files = []
hgrepos = []
issue_num = 30004
keywords = ['patch']
message_count = 13.0
messages = ['291209', '291246', '292511', '292972', '306361', '306439', '306440', '306447', '306942', '306943', '306944', '306945', '306947']
nosy_count = 8.0
nosy_names = ['akuchling', 'ezio.melotti', 'mrabarnett', 'docs@python', 'serhiy.storchaka', 'Mariatta', 'mandeepb', 'Cristian Barbarosie']
pr_nums = ['4443', '4554', '4555']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue30004'
versions = ['Python 2.7', 'Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by fc9cca60-f582-41bb-a683-840c6b119ece** [comment]
In the Regular Expression HOWTO
https://docs.python.org/3.6/howto/regex.html#regex-howto
the last example in the "Grouping" section has a bug. The code is supposed to find repeated words, but it catches false repetitions.

```pycon
>>> p = re.compile(r'(\b\w+)\s+\1')
>>> p.search('Paris in the the spring').group()
'the the'
>>> p.search('k is the thermal coefficient').group()
'the the'
```


I propose adding a \\b after \\1, this solves the problem :

```pycon
>>> p = re.compile(r'(\b\w+)\s+\1\b')
>>> p.search('Paris in the the spring').group()
'the the'
>>> print p.search('k is the thermal coefficient')
None
```

**Comment by fc9cca60-f582-41bb-a683-840c6b119ece** [comment]
Just discovered that a nearly identical example is presented in the end of section "Non-capturing and Named Groups". My proposal applies to this other example, too.
And, by the way, reading this HOWTO has been very useful to me.

**Comment by aa752287-3877-4488-b28e-6a0aef4e10d3** [comment]
Looks like both examples need a closing \\b. Is this being worked on or should I submit a PR?

**Comment by fc9cca60-f582-41bb-a683-840c6b119ece** [comment]
This topic seems stuck. Is there anything else I should do ?

**Comment by serhiy-storchaka** [comment]
Do you mind to create a pull request on GitHub Cristian?

**Comment by fc9cca60-f582-41bb-a683-840c6b119ece** [comment]
I'm sorry, I have no experience at all with Git. Could you please do it for me ?
The bug appears in two places, see my first two messages.
Thank you

**Comment by aa752287-3877-4488-b28e-6a0aef4e10d3** [comment]
Serhiy, Christian: I'll submit a PR for this later today.

**Comment by aa752287-3877-4488-b28e-6a0aef4e10d3** [comment]
Cristian, Serhiy: I've submitted a PR for this bug. 

Cristian: I apologize for misspelling your name in a prior post.

**Comment by Mariatta** [comment]
New changeset 610e5afdcbe3eca906ef32f4e0364e20e1b1ad23 by Mariatta (Mandeep Bhutani) in branch 'master':
bpo-30004: Fix the code example of using group in Regex Howto Docs (GH-4443)
https://github.com/python/cpython/commit/610e5afdcbe3eca906ef32f4e0364e20e1b1ad23

**Comment by Mariatta** [comment]
New changeset c02037d62284f4d4ca6b22f2ed05165ce2014951 by Mariatta (Miss Islington (bot)) in branch '2.7':
bpo-30004: Fix the code example of using group in Regex Howto Docs (GH-4443) (GH-4555)
https://github.com/python/cpython/commit/c02037d62284f4d4ca6b22f2ed05165ce2014951

**Comment by Mariatta** [comment]
New changeset 3e60747025edc34b503397ab8211be59cfdd05cd by Mariatta (Miss Islington (bot)) in branch '3.6':
bpo-30004: Fix the code example of using group in Regex Howto Docs (GH-4443) (GH-4554)
https://github.com/python/cpython/commit/3e60747025edc34b503397ab8211be59cfdd05cd

**Comment by Mariatta** [comment]
Thanks everyone. I merged the PR, and it's been backported to 3.6 and 2.7

**Comment by serhiy-storchaka** [comment]
Thank you Cristian for reporting this issue. Thank you Mandeep for your patch. Thank you Mariatta for merging.
